node/storage-cleaner/src/appwrite.js

```diff
@@ -0,0 +1,98 @@
+import { Client, Storage, Query } from 'node-appwrite';
+import { getExpiryDate } from './util.js';
+
+class AppwriteService {
+  defaultQuery = [Query.orderAsc('$createdAt')];
+
+  constructor() {
+    const client = new Client()
+      .setEndpoint(
+        process.env.APPWRITE_ENDPOINT ?? 'https://cloud.appwrite.io/v1'
+      )
+      .setProject(process.env.APPWRITE_FUNCTION_PROJECT_ID)
+      .setKey(process.env.APPWRITE_API_KEY);
+    this.storage = new Storage(client);
+  }
+
+  /**
+   * Clean up all storage buckets by removing files older than a specified retention period.
+   *
+   * This function retrieves a list of all storage buckets, and for each bucket,
+   * it calls the cleanBucket function to remove files that are older than the calculated expiry date
+   * based on the retention period.
+   *
+   * @returns {Promise<void>} A Promise that resolves when all buckets are cleaned.
+   */
+  async cleanAllBuckets() {
+    const buckets = await this.getBuckets();
+    for (const bucket of buckets) {
+      await this.cleanBucket(bucket.$id);
+    }
+  }
```

Comment:
As comented in README, let's only clean one bucket based on ENV variable.

Reply:
We can probably remove the whole `getBuckets` as well.

Reply:
We likely also dont need `defaultQuery` anymore

Reply:
Done, the do while change is good, it made the code smaller.